indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -179,11 +192,20 @@ public int compare(Pair<Task, TaskLock> left, Pair<Task, TaskLock> right)
    */
   public TaskLock lock(final Task task, final Interval interval) throws InterruptedException
   {
+    long startTime = System.currentTimeMillis();
     giant.lock();
     try {
       Optional<TaskLock> taskLock;
       while (!(taskLock = tryLock(task, interval)).isPresent()) {
-        lockReleaseCondition.await();
+        lockReleaseCondition.await(lockTimeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
If this line returns early that means Task got the lock, not sure why final variable of this class should be updated ?